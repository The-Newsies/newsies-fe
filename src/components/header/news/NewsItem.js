import React from 'react';
import PropTypes from 'prop-types';

export default function NewsItem({ 
  title, 
  description, 
  date, 
  author, 
  sourceLink, 
  sourceName, 
  image,
  tagColor }) {
  return (
    <div>
      <div><h3>{title}</h3></div>
      <div><p>{description}</p></div>
      <div><p>{author}</p></div>
      <div><p>{date}</p></div>
      <div><img src={image} /></div>
      <div><a href={sourceLink}>{sourceName}</a></div>
      <div><div>{tagColor}</div></div>
    </div>
  );
}

NewsItem.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  sourceName: PropTypes.string.isRequired,
  sourceLink: PropTypes.string.isRequired,
  tagColor: PropTypes.string.isRequired
};
