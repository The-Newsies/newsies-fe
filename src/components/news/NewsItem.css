.container {
  display: flex;
  align-items: center;
  border: solid black 1px;
  height: 168px;
  width: 1100px;
}

.container img {
  height: 169px;
}

.imageContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.imageContainer p {
  margin-left: 3px;
}

.titleContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 15px;
  width: 650px;
}

.links {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

