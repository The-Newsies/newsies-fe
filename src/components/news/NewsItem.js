import React from 'react';
import PropTypes from 'prop-types';

export default function NewsItem({ 
  title, 
  description, 
  date, 
  author, 
  sourceLink, 
  sourceName, 
  image,
  tagColor }) {

  const tagColorStyles = {
    background: tagColor,
    width: '100x',
    height: '100px',
  };

  return (
    <div>
      <div><h3>{title}</h3></div>
      <div><p>{description}</p></div>
      <div><p>Author: {author}</p></div>
      <div><p>Published: {date}</p></div>
      <div><img src={image} /></div>
      <div><a href={sourceLink}>Full Article at {sourceName}</a></div>
      <div><div style={tagColorStyles}></div></div>
    </div>
  );
}

NewsItem.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  sourceName: PropTypes.string.isRequired,
  sourceLink: PropTypes.string.isRequired,
  tagColor: PropTypes.string.isRequired
};
