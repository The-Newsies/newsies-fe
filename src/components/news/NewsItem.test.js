import React from 'react';
import { shallow } from 'enzyme';
import NewsItem from './NewsItem';

describe('NewsItem component', () => {
  it('renders NewsItem', () => {
    const wrapper = shallow(<NewsItem 
      title={'title'}  
      description={'description'}  
      publishedAt={'publishedAt'}
      author={'author'}
      url={'url'}
      source={'source'}
      urlToImage={'urlToImage'}
      tagColor={'tagColor'}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
