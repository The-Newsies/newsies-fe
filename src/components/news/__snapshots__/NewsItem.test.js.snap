// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem component renders NewsItem 1`] = `
<div
  className="container"
>
  <div
    className="linkContainer"
  >
    <a
      href="url"
      rel="noopener noreferrer"
      target="_blank"
    >
      View at 
    </a>
    <button>
      âž•
    </button>
  </div>
  <div
    className="imageContainer"
  >
    <img
      src="urlToImage"
    />
  </div>
  <div
    className="titleContainer"
  >
    <h3>
      title
    </h3>
    <p
      className="title"
    >
      description
    </p>
    <footer
      className="source"
    >
      <p>
        Published: 
        publishedAt
      </p>
      <p>
        Author: 
        author
      </p>
    </footer>
  </div>
</div>
`;
